src/utils/profileLink.ts

```diff
@@ -0,0 +1,134 @@
+/*
+* This program and the accompanying materials are made available under the terms of the *
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
+* https://www.eclipse.org/legal/epl-v20.html                                      *
+*                                                                                 *
+* SPDX-License-Identifier: EPL-2.0                                                *
+*                                                                                 *
+* Copyright Contributors to the Zowe Project.                                     *
+*                                                                                 *
+*/
+
+import * as readYaml from "js-yaml";
+import * as writeYaml from "yamljs";
+import * as vscode from "vscode";
+import { IProfileLoaded, Logger } from "@zowe/imperative";
+import * as path from "path";
+import * as fs from "fs";
+import { Profiles } from "../Profiles";
+import { ZoweTreeNode } from "../abstract/ZoweTreeNode";
+import { IZoweTreeNode } from "../api/IZoweTreeNode";
+import { getZoweDir } from "../utils";
+import * as nls from "vscode-nls";
+const localize = nls.config({ messageFormat: nls.MessageFormat.file })();
+
+const LINKS_FOLDER = "profile_links";
+const FILE_SUFFIX = ".yaml";
+
+export async function getLinkedProfile(node: IZoweTreeNode, type: string, logger?: Logger) {
```

Comment:
Functional vs OO? 
I believe we talked about this, but I don't want to disrupt the implementation too much (and I don't have a strong opinion in this scenario to go OO here, other than what we have discussed in the past)

Reply:
I didn't really feel it was an OO concept and that it was simpler doing it this way. Could have used a singleton but just felt right like this as a support function rather than class